src/user/user.service.ts

```diff
@@ -13,7 +13,14 @@ export class UserService {
         @InjectModel(User) private readonly model: ReturnModelType<typeof User>,
     ) {}
 
-    find(): Promise<User[]> {
+    find(role?: UserRole): Promise<User[]> {
+        if (role) {
+            return this.model
+                .find({
+                    role: role,
+                })
```

Comment:
just `{ role }` is enough since `key` and `value` of the object is exactly the same

Reply:
`find()` should accept any filters for further usage i.e. `find(filters)`.

Reply:
Roger that I will change word to be filter